src/main/java/ladder/domain/ladder/Ladder.java

```diff
@@ -6,17 +6,48 @@
 import ladder.domain.ladder.generator.RungGenerator;
 
 public class Ladder {
-    private final List<Line> lines;
+    private static final int MINIMUM_HEIGHT = 1;
+    private static final int MAXIMUM_HEIGHT = 30;
 
-    public Ladder(int playerCount, LadderHeight ladderHeight, RungGenerator rungGenerator) {
-        this.lines = generateLines(playerCount, ladderHeight.getHeight(), rungGenerator);
+    private final List<Line> lines;
 
+    private Ladder(List<Line> lines) {
+        this.lines = lines;
     }
 
-    private List<Line> generateLines(int playerCount, int height, RungGenerator rungGenerator) {
-        return Stream.generate(() -> new Line(playerCount, rungGenerator))
+    public static Ladder of(int height, int playerCount, RungGenerator rungGenerator) {
+        validateHeightRange(height);
+
+        List<Line> lines = Stream.generate(() -> new Line(playerCount, rungGenerator))
                 .limit(height)
                 .toList();
+
+        return new Ladder(lines);
+    }
+
+    private static void validateHeightRange(int height) {
+        if (MAXIMUM_HEIGHT < height || height < MINIMUM_HEIGHT) {
```

Comment:
(초미세 가독성 훈수 2탄)
이건 저만 그런걸수도 있는데 최소 표현부터 먼저하고 최대 표현을 나중에 해야 잘 읽히는거 같아요 -.-ㅎ 
```suggestion
        if (height < MINIMUM_HEIGHT || MAXIMUM_HEIGHT < height) {
```
비교해보시고 그냥 구름이 잘 읽힌다 싶은걸로 해주세용

Reply:
음 height를 안으로 모아주고 싶었어요!

아래와 같이 최소를 먼저 표현하는걸로 변경했어요.

```java
if (MINIMUM_HEIGHT > height || height > MAXIMUM_HEIGHT) 
```

혹시 부등호 방향도 정해진 기준이 있나요? "<" 가 읽기 쉽다던지 ">"가 읽기 쉽다던지

Reply:
아뇨 이런 부분은 좀 느낌에 가까운거라서요 ㅋ ㅋ 개인적으로 잘 읽힌다고 생각하시는 걸 쓰시면 됩니다 (사실 우린 정확히 남 입장이 될 수 없으니까요, 그냥 "내가 이 코드를 처음 보는 사람이라면 어떨까?" 하고 상상하면서 짜는거죠. 어쨌든 스스로 타자화 해보는 것도 한계는 있죠)

매직넘버 상수화 같은건 전 인류 공통의 가독성 증진으로 봐도 좋은데 
이런 부분 가독성은 문화권 마다 다를거에요.

저는 가독성과 관련된 개인적인 입장 정리는 [프로그래머의 뇌](https://www.yes24.com/Product/Goods/105911017?pid=123487&cosemkid=go16418003457693686&gad_source=1&gclid=Cj0KCQiAxOauBhCaARIsAEbUSQSO68qZKzO1kGO7NvKygKJgWLwrzGmLms2AlMEWg4Hc9aItamdkBx4aAo4DEALw_wcB) 읽어보면서 좀 됐어요~ 
가독성의 이론적 배경을 알려주니까요, 심심할 때 한번 가벼운 마음으로 읽어보세요